numpy/core/tests/test_scalarbuffer.py

```diff
@@ -0,0 +1,96 @@
+"""
+Test scalar buffer interface adheres to PEP 3118
+"""
+import sys
+import numpy as np
+from numpy.testing import run_module_suite, assert_, dec
+
+# types
+scalars = [
+    np.bool_, np.bool8, np.byte, np.short, np.intc, np.int_,
+    np.longlong, np.intp, np.int8, np.int16, np.int32, np.int64,
+    np.ubyte, np.ushort, np.uintc, np.uint, np.ulonglong,
+    np.uintp, np.uint8, np.uint16, np.uint32, np.uint64,
+    np.half, np.single, np.double, np.float_, np.longfloat,
+    np.float16, np.float32, np.float64, np.csingle, np.complex_,
+    np.clongfloat, np.complex64, np.complex128,
+]
```

Comment:
This list is longer than it needs to be, as most of these are aliases. I'd recommend testing only:

```python
[
    np.bool,
    np.byte, np.short, np.intc, np.int_, np.longlong,
    np.ubyte, np.ushort, np.uintc, np.uint, npu.longlong,
    np.half, np.single, np.double, np.longdouble,
    np.csingle, np.cdouble, np.clongdouble,
]
```